contracts/schemes/DxDaoSchemeConstraints.sol

```diff
@@ -0,0 +1,149 @@
+        // constraint approve calls
+            if (callData[0] == APPROVE_SIGNATURE[0] &&
+                callData[1] == APPROVE_SIGNATURE[1] &&
+                callData[2] == APPROVE_SIGNATURE[2] &&
+                callData[3] == APPROVE_SIGNATURE[3]) {
+                uint256 amount;
+                address contractToCall = _contractsToCall[i];
+                // solhint-disable-next-line no-inline-assembly
+                assembly {
+                    amount := mload(add(callData, 68))
+                }
+                periodSpendingToken[observervationIndex][contractToCall] =
+                periodSpendingToken[observervationIndex][contractToCall].add(amount);
+                require(
+                periodSpendingToken[observervationIndex][contractToCall] <= periodLimitToken[contractToCall],
+                "periodSpendingTokensExceeded");
+            }
+
+        }
+        periodSpendingWei[observervationIndex] =
+        periodSpendingWei[observervationIndex].add(totalPeriodSpendingInWei);
+        require(periodSpendingWei[observervationIndex] <= periodLimitWei, "periodSpendingWeiExceeded");
+        return true;
+    }
+
+    /*
+     * @dev isAllowedToPropose should be called upon a proposal submition.
+     * allow only whitelisted target contracts or 'approve' calls which the 'spender' is whitelisted
+     * @param _contractsToCall the contracts to be called
+     * @param _callsData - The abi encode data for the calls
+     * @param _values value(ETH) to transfer with the calls
+     * @param _avatar avatar
+     * @return bool value true-allowed false not allowed
+     */
+    function isAllowedToPropose(
+        address[] calldata _contractsToCall,
+        bytes[] calldata _callsData,
+        uint256[] calldata,
+        Avatar)
+    external
+    returns(bool)
+    {
+        for (uint i = 0; i < _contractsToCall.length; i++) {
+            if (!contractsWhiteListMap[_contractsToCall[i]]) {
+                address spender;
+                bytes memory callData = _callsData[i];
+                require(
+                    callData[0] == APPROVE_SIGNATURE[0] &&
+                    callData[1] == APPROVE_SIGNATURE[1] &&
+                    callData[2] == APPROVE_SIGNATURE[2] &&
+                    callData[3] == APPROVE_SIGNATURE[3],
+                "allow only approve call for none whitelistedContracts");
+                //in solidity > 6 this can be replaced by:
+                //(spender,) = abi.descode(callData[4:], (address, uint));
+                // see https://github.com/ethereum/solidity/issues/9439
+                // solhint-disable-next-line no-inline-assembly
+                assembly {
+                    spender := mload(add(callData, 36))
+                }
+                require(contractsWhiteListMap[spender], "spender contract not whitelisted");
+            }
+        }
+        return true;
+    }
+
+    function observationIndex() public view returns (uint256) {
```

Comment:
@orenyodfat this could become vulnerable to overflows if the period is set to a super low level, I would propose to add a require in the initialize() to set the period to a min limit as of 7days.